services/src/main/java/org/keycloak/services/resources/ThemeResource.java

```diff
@@ -107,7 +108,9 @@ public Response getResource(@PathParam("version") String version, @PathParam("th
     @Produces(MediaType.APPLICATION_JSON)
     public Response getLocalizationTexts(@PathParam("realm") String realmName, @QueryParam("theme") String theme,
                                          @PathParam("locale") String localeString, @PathParam("themeType") String themeType,
-                                         @QueryParam("source") boolean showSource) throws IOException {
+                                         @QueryParam("source") boolean showSource, @QueryParam("hasWords") List<String> hasWords,
```

Comment:
I'd suggest changing the `@QueryParam("hasWords")` to `@QueryParam("q")`  and changing its type to `String` similarly to other free-text search fields in REST API, e.g. [here](https://github.com/keycloak/keycloak/blob/e13d3264a2c770f390460bf0db9247e1bf77646a/services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java#L404) or [here](https://github.com/keycloak/keycloak/blob/e13d3264a2c770f390460bf0db9247e1bf77646a/services/src/main/java/org/keycloak/services/resources/admin/GroupsResource.java#L88).

Then it would be possible to process the `q` field into a `Set<String>` which would contain non-empty and non-null elements trimmed of whitespace. Having a `Set` instead of `List` would allow simplification of the condition on L 146.

Reply:
These examples the searchQuery has the form `name:value`, but here it's just a number of words that have to be in the message bundle these are no attribute names.